### PR TITLE
Add faktory-cli command to generate a local self signed certificate

### PR DESCRIPTION
**Pull request by cdrx**
You can supply a hostname, or if left blank will default to a wildcard certificate that matches all hostnames

### CODE REVIEW

**Comment by mperham** [comment]
Awesome start but I'm not sure I like the implicit dependency on openssl.  I found this which is Go-native, perhaps we could adapt that?

https://golang.org/src/crypto/tls/generate_cert.go

It would be great to have the Docker image automatically provision itself with a cert.  Any ideas on how best to tie in with Docker's TLS certs?

**Comment by cdrx** [comment]
That looks better because it can generate certs for an IP address, I'll replace the shelled out call to openssl with that.

It will be easy to write an entrypoint script for the docker container to generate some certs when it starts, but first we would need to be able to use `faktory-cli` non-interactively -- it needs to be able to be scriptable, like:

```
$ faktory-cli generate-certificate hostname
```

**Comment by mperham** [comment]
Yep I'm on board with CLI exiting immediately if the ARGs supply all necessary data.  Should be interactive only if run with no args.

**Comment by cdrx** [comment]
@mperham this may be a little redundant now ;) but I've updated this to use go's crypto lib

File: cmd/faktory-cli/repl.go [hunk 1]
```diff
@@ -178,6 +196,96 @@ func restore(store storage.Store) error {
...
+	if ok {
+		return fmt.Errorf("public.crt already exists in %v, will not create a new one", path)
+	}
+
+	ok, _ = util.FileExists(private)
+	if ok {
+		return fmt.Errorf("private.key already exists in %v, will not create a new one", path)
+	}
+
+	if len(hostnames) == 0 {
```
Comment (mperham) [inline cmd/faktory-cli/repl.go]: You fill in "*" by default so this error will never happen.

**Comment by mperham** [comment]
So, with the upcoming removal of TLS, is this functionality better served as a wiki doc on how to generate a cert and configure stunnel in front of Faktory?

**Comment by cdrx** [comment]
Yep! Don’t think there is much point to this functionality now

**Pull request closed without merging**